src/lib.rs

```diff
@@ -450,13 +638,19 @@ impl<T> Grid<T> {
     ///
     /// Panics if the col index is out of bounds.
     pub fn iter_col(&self, col: usize) -> StepBy<Iter<T>> {
```

Comment:
In the case of `iter_col`, I think column-major will be faster, since values will be adjacent, but if your point is that it should be documented as well, I agree.

I added performance notes on `iter_col`, `iter_col_mut`, `iter_row` and `iter_row_mut`.